<?xml version="1.0" ?>
<launch>
    <![CDATA[
    <!--
          FOR DEBUGGING: add to node tag: launch-prefix="gnome-terminal -x gdb --args"
    -->
    ]]>

    <!-- start standalone node -->
    <node
          pkg="lidar_pose_estimation"
          name="lidar_pose_estimation"
          exec="lidar_pose_estimation_node"
    >
        <param name="max_dist" value="$(env ROS_LIDAR_CALIBRATION_MAX_DISTANCE 0.25)" />
        <param name="execution_frequency" value="$(env ROS_LIDAR_CALIBRATION_EXECUTION_FREQUENCY 1.0)" />
        <param name="cluster_tolerance" value="$(env ROS_LIDAR_CALIBRATION_CLUSTER_TOLERANCE 0.02)" />
        <param name="min_cluster_size" value="$(env ROS_LIDAR_CALIBRATION_MIN_CLUSTER_SIZE 10)" />
        <param name="max_cluster_size" value="$(env ROS_LIDAR_CALIBRATION_MAX_CLUSTER_SIZE 100)" />
        <param name="max_reference_distance_deviation" value="$(env ROS_LIDAR_CALIBRATION_MAX_REFERENCE_DEVIATION 0.01)" />
        <param name="roll" value="$(env ROS_LIDAR_CALIBRATION_ROLL 0.0)" />
        <param name="pitch" value="$(env ROS_LIDAR_CALIBRATION_PITCH 0.0)" />
        <param name="z" value="$(env ROS_LIDAR_CALIBRATION_Z 0.0)" />
        <param name="p_ref_1_x" value="$(env ROS_LIDAR_CALIBRATION_P_REF_1_X -0.035)" />
        <param name="p_ref_1_y" value="$(env ROS_LIDAR_CALIBRATION_P_REF_1_Y -0.055)" />
        <param name="p_ref_2_x" value="$(env ROS_LIDAR_CALIBRATION_P_REF_2_X -0.035)" />
        <param name="p_ref_2_y" value="$(env ROS_LIDAR_CALIBRATION_P_REF_2_Y -0.055)" />
    </node>
</launch>
